src/wp-admin/includes/class-wp-debug-data.php

```diff
@@ -243,6 +243,13 @@ public static function debug_data() {
 			$wp_environment_type = __( 'Undefined' );
 		}
 
+		// Check WP_DEVELOPMENT_MODE.
+		if ( defined( 'WP_DEVELOPMENT_MODE' ) && WP_DEVELOPMENT_MODE ) {
+			$wp_development_type = WP_DEVELOPMENT_MODE;
+		} else {
+			$wp_development_type = __( 'Undefined' );
+		}
```

Comment:
Not sure the `if` / `else` is needed? The constant is always defined, as it's set in `wp_initial_constants()`.